### DISCUSSION TITLE
Connect switches

### DISCUSSION THREAD

**Discussion by RoyMagne** [plamish/xcomfort#17] [Q&A]
Hi.

Thank you for making this integration for Home Assistant.
I have just started using it, and all my dimming and light actuators are working fine, and great to finaly be able to integrate my Xcomfort system into Home Assistant.

I have made an own zone with all my lights and all imported fine, but when I tried to add a switch to this zone it does not import.
What could be the reason for this? Are switches not supported?

**Comment by plamish** [comment]
Hey RoyMagne, 
Are we talking about switch actuators or about wall switches?

Switching actuators are supported, but they need to be defined as general in SHC.

Wall switches are not supported in the integration. In this issue you can find why they are not supported and how xcomfort2mqtt broker can help with having them working in HA:
https://github.com/plamish/xcomfort/issues/11

**Reply by RoyMagne** [reply]
Hi Plamish.

Thank you for your quick reply. 
I have installed xcomfort2mqtt and I have set up the switches. Thank you :)

It would be nice if they could be added to the integreation, but are there any limitations preventing you to include them?

**Comment by plamish** [comment]
Today the integration pulls statuses every few seconds from SHC meaning there would be a need to keep switch/button pressed for at least few seconds to ensure its state is captured in HA.

xcomfort2mqtt uses polling technique to get the statuses right after an event happens.

I plan to rewrite the integration to leverage polling mechanism and include wall switches. Stay tuned

**Reply by RoyMagne** [reply]
Thank you for making this integration, and I really look forward to future updates.
